include/dxc/Support/Path.h

```diff
@@ -0,0 +1,128 @@
+///////////////////////////////////////////////////////////////////////////////
+//                                                                           //
+// Path.h                                                                    //
+// Copyright (C) Microsoft Corporation. All rights reserved.                 //
+// This file is distributed under the University of Illinois Open Source     //
+// License. See LICENSE.TXT for details.                                     //
+//                                                                           //
+// Helper for HLSL related file paths.                                       //
+//                                                                           //
+///////////////////////////////////////////////////////////////////////////////
+#pragma once
+
+#include "llvm/ADT/StringRef.h"
+#include <string>
+
+namespace hlsl {
+
+template <typename CharTy>
+bool IsAbsoluteOrCurDirRelativeImpl(const CharTy *Path) {
+  if (!Path || !Path[0])
+    return false;
+  // Current dir-relative path.
+  if (Path[0] == '.') {
+    return Path[1] == '\0' || Path[1] == '/' || Path[1] == '\\';
+  }
+  // Disk designator, then absolute path.
+  if (Path[1] == ':' && (Path[2] == '\\' || Path[2] == '/')) {
```

Comment:
In the case that the path passed is malformed, like "C", Path[1] would be an out of bounds access at this point. Would we want to check that Path[1] exists first? Likewise for Path[2]

Reply:
Since "C" is null-terminated, Path[1] would be '\0' and the subsequent Path[2] would not be checked. Although this is pretty confusing and unsafe... So I should probably change it.

Reply:
I made it take a length and check the length instead.